File: euler010.py
"""
A solution to Project Euler Problem 10.

https://projecteuler.net/problem=10
"""

import itertools

from common import primes

def sum_of_primes_below(n: int) -> int:
    """
    Returns the sum of all of the prime integers that are less than n.

    Example:
    >>> sum_of_primes_below(10)
    17
    """
    return sum(itertools.takewhile(lambda x: x < n, primes()))

if __name__ == '__main__':
    print(sum_of_primes_below(2000000)) # produces 142913828922

File: haskell/Euler010.hs
module Euler010 where

import Common (primes)

-- | Search cap for prime sumation
target :: Int
target = 2000000

-- | Computes the sum of all primes below the given integer.
sumOfPrimesBelow :: Int -> Int
sumOfPrimesBelow cap = sum . takeWhile (<cap) $ primes

main :: IO ()
main = print $ sumOfPrimesBelow target
